File: bench/ref/matmul.js
function matmul(n){ const a=new Int32Array(n*n),b=new Int32Array(n*n),c=new Int32Array(n*n);
  for(let i=0;i<n;i++) for(let j=0;j<n;j++){ a[i*n+j]=(i+j)%100; b[i*n+j]=(i*j)%100; }
  for(let i=0;i<n;i++) for(let j=0;j<n;j++){ let s=0; for(let k=0;k<n;k++) s+=a[i*n+k]*b[k*n+j]; c[i*n+j]=s; }
  let sum=0; for(let i=0;i<n*n;i++) sum=(sum+c[i])%1000000007;
  return sum; }
console.log(matmul(400));

File: bench/ref/matmul.rs
fn matmul(n:usize)->i64{ let mut a=vec![0i64;n*n]; let mut b=vec![0i64;n*n]; let mut c=vec![0i64;n*n];
  for i in 0..n { for j in 0..n { a[i*n+j]=((i+j)%100) as i64; b[i*n+j]=((i*j)%100) as i64; } }
  for i in 0..n { for j in 0..n { let mut s=0i64; for k in 0..n { s+=a[i*n+k]*b[k*n+j]; } c[i*n+j]=s; } }
  let mut sum=0i64; for i in 0..n*n { sum=(sum+c[i])%1000000007; } sum }
fn main(){ println!("{}", matmul(400)); }
